lsp/nls/src/linearization/building.rs

```diff
@@ -32,6 +32,26 @@ impl<'b> Building<'b> {
         self.linearization.push(item)
     }
 
+    fn get_item_kind_with_id(&self, file: FileId, id: ItemId) -> Option<(&ItemId, &TermKind)> {
+        if file == id.file_id {
+            // This usage references an item in the file we're currently linearizing
+            let item = self.linearization.get(id.index)?;
+            Some((&item.id, &item.kind))
+        } else {
+            // This usage references an item in another file (that has already been linearized)
+            let item = self
+                .lin_cache
+                .get(&id.file_id)?
+                .get_item(id, self.lin_cache)?;
+            Some((&item.id, &item.kind))
+        }
+    }
+
+    fn get_item_kind(&self, file: FileId, id: ItemId) -> Option<&TermKind> {
+        let (_, kind) = self.get_item_kind_with_id(file, id)?;
+        Some(kind)
+    }
+
```

Comment:
>Would it make sense to rather have a `get_item` and `get_item_mut` functions, as in `Completed`, rather than those specialized versions? 

That would have been better, but the problem with it is that, with `Building`, we're not yet done linearizing the current file, so the linearization items in `Building` have the type `LinearizationItem<UnifType>`, while the linearization items in the other files would have the type `LinearizationItem<Types>`, so that will lead to a return type mismatch from the function. I noticed that we only need the `TermKind` (and in some cases, the `ItemId`), so this was a nice way to avoid using the whole struct regardless of the generic parameter.

Reply:
>Then the caller could just get the item and decide what to do with it. Here it feels a bit like we are just instantiating a get_item function with some arbitrary post-processing.

Right. It's just a convenience, because we know the caller does just one thing it. But we can change it to avoid the "post-processing"